fidimag/common/sundials/cvode.pyx

```diff
@@ -246,4 +301,194 @@ cdef class CvodeSolver(object):
         self.user_data.v = NULL
         self.user_data.jv = NULL
         N_VDestroy_Serial(self.u_y)
-        CVodeFree(&self.cvode_mem)
+        CVodeFree(& self.cvode_mem)
+
+cdef class CvodeSolver_OpenMP(object):
+    cdef public double t
+    cdef public np.ndarray y
+    cdef double rtol, atol
+    cdef int cvode_already_initialised
+    cdef np.ndarray y0
+    cdef np.ndarray dm_dt
+    cdef np.ndarray mp
+    cdef np.ndarray Jmp
+    cdef N_Vector u_y
+    cdef void * cvode_mem
+    cdef void * rhs_fun
+    cdef void * jvn_fun
+    cdef callback_fun
+    cdef jtimes_fun
+    cdef cv_userdata user_data
+    cdef long int nsteps, nfevals, njevals
+    cdef int max_num_steps
+    cdef int has_jtimes
+    cdef str linear_solver
+    cdef str parellel_solver
+    cdef int num_threads
+
+    def __cinit__(self, spins, rhs_fun, jtimes_fun=None, linear_solver="spgmr", rtol=1e-8, atol=1e-8):
+        if os.environ['OMP_NUM_THREADS']:
+            try:
+                self.num_threads = int(os.environ['OMP_NUM_THREADS'])
```

Comment:
Alternatively, omp_get_max_threads() can be used to get the number of threads set by OMP_NUM_THREADS.